### ISSUE TITLE
bug/get_word_bounding_box_from_element function throws "UnboundLocalError: cannot access local variable 'isalnum' where it is not associated with a value" exception

### ISSUE THREAD

**Issue by AC15** [Unstructured-IO/unstructured#1778]
**Describe the bug**
The "get_word_bounding_box_from_element" function from the "unstructured/partition/pdf.py" file throws "UnboundLocalError: cannot access local variable 'isalnum' where it is not associated with a value" exception on line 1774, due to the isalnum variable not being defined when the if statement on line 1172 is not triggered.

```python
                # TODO(klaijan) - isalnum() only works with A-Z, a-z and 0-9
                # will need to switch to some pattern matching once we support more languages
                if not word:
                    isalnum = char.isalnum()
                if word and char.isalnum() != isalnum:
                    isalnum = char.isalnum()
                    words.append(
                        {"text": word, "bbox": (x1, y1, x2, y2), "start_index": start_index},
                    )
                    word = ""
```

A direct link to the line:
https://github.com/Unstructured-IO/unstructured/blob/0.10.24/unstructured/partition/pdf.py#L1174

**Additional context**
A part of the exception:
```
UnboundLocalError: cannot access local variable 'isalnum' where it is not associated with a value
Traceback:
File "C:\Users\REDACTED\AppData\Roaming\Python\Python311\site-packages\langchain\document_loaders\s3_directory.py", line 135, in load
    docs.extend(loader.load())
                ^^^^^^^^^^^^^
File "C:\Users\REDACTED\AppData\Roaming\Python\Python311\site-packages\langchain\document_loaders\unstructured.py", line 86, in load
    elements = self._get_elements()
               ^^^^^^^^^^^^^^^^^^^^
File "C:\Users\REDACTED\AppData\Roaming\Python\Python311\site-packages\langchain\document_loaders\s3_file.py", line 126, in _get_elements
    return partition(filename=file_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\auto.py", line 316, in partition
    elements = _partition_pdf(
               ^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\documents\elements.py", line 276, in wrapper
    elements = func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\file_utils\filetype.py", line 551, in wrapper
    elements = func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\chunking\title.py", line 211, in wrapper
    elements = func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\pdf.py", line 148, in partition_pdf
    return partition_pdf_or_image(
           ^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\pdf.py", line 245, in partition_pdf_or_image
    extracted_elements = extractable_elements(
                         ^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\pdf.py", line 171, in extractable_elements
    return _partition_pdf_with_pdfminer(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\utils.py", line 159, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\pdf.py", line 423, in _partition_pdf_with_pdfminer
    elements = _process_pdfminer_pages(
               ^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\pdf.py", line 507, in _process_pdfminer_pages
    _, words = get_word_bounding_box_from_element(obj, height)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Program Files\Python311\Lib\site-packages\unstructured\partition\pdf.py", line 1004, in get_word_bounding_box_from_element
    if char.isalnum() != isalnum:
                         ^^^^^^^
```

**Comment by Klaijan** [comment]
Hi @AC15, can you make sure you use the latest release of `unstructured`? the bug is fixed on the release version 0.10.19.

**Comment by AC15** [comment]
Hi @Klaijan, just tested it the newer version worked. Thank you.
